src/puppetlabs/puppetdb/metrics/core.clj

```diff
@@ -30,60 +31,21 @@
               [k (str v)]))))
 
 (defn all-mbean-names
-  "Return a set of all mbeans names"
+  "Return a seq of all mbeans names"
   []
-  {:post [(set? %)]}
-  (set (map str (jmx/mbean-names "*:*"))))
-
-(defn linkify-names
-  "Return a map of mbean name to a link that will retrieve the
-  attributes"
-  [names]
-  (zipmap names (map #(format "/mbeans/%s" (url-encode %)) names)))
+  {:post [(coll? %)]}
+  (map str (jmx/mbean-names "*:*")))
 
 (defn mbean-names
-  "Returns a JSON array of all MBean names"
-  [_]
+  "Return a map of mbean name to a link that will retrieve the attributes"
+  []
   (->> (all-mbean-names)
-       (linkify-names)
-       (into (sorted-map))
-       (http/json-response)))
+       (map #(vector % (format "/mbeans/%s" (url-encode %))))
+       (into (sorted-map))))
 
 (defn get-mbean
   "Returns the attributes of a given MBean"
   [name]
-  (if ((all-mbean-names) name)
```

Comment:
This refactoring wasn't all 100% necessary, but I was already refactoring parts of it to remove some of the namespace cyclic dependencies I was getting by putting the metrics-registry in this namespace, so I thought it couldn't hurt to tidy this code up a bit.